export/kafka.go

```
@@ -0,0 +1,200 @@
+package export
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"go.opencensus.io/metric/metricdata"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"google.golang.org/protobuf/proto"
+)
+
+var (
+	messagesSent    = stats.Int64("messages_sent", "the number of metric messages successfully sent", "1")
+	messagesDropped = stats.Int64("messages_dropped", "the number of metric messages dropped", "1")
+)
+
+var (
+	messagesSentView = &view.View{
+		Name:        "messages_sent",
+		Measure:     messagesSent,
+		Description: "the number of metric messages successfully sent",
+		Aggregation: view.Count(),
+	}
+
+	messagesDropedView = &view.View{
+		Name:        "messages_dropped",
+		Measure:     messagesDropped,
+		Description: "the number of metric messages dropped",
+		Aggregation: view.Sum(),
+	}
+)
+
+// TopicConfig holds the configurations for Topic info
+type TopicConfig struct {
+	Topic         string
+	NumPartitions int
+	NumReplicas   int
+}
+
+// Kafka is an exporter that exports metrics to a
+// Kafka broker.
+type Kafka struct {
+	config              Config
+	kafkaConfig         *kafka.ConfigMap
+	producer            *kafka.Producer
+	topicInfo           TopicConfig
+	messageFlushTimeSec int
+	lastDroppedLogCount int
+}
+
+// NewKafka returns a new Kafka exporter
+func NewKafka(config Config, kafkaConfig *kafka.ConfigMap, topicInfo TopicConfig) *ExporterAgent {
+	if err := view.Register(messagesSentView, messagesDropedView); err != nil {
+		log.Printf("Problem registering views: %v", err)
```

should return error, rather than continuing